src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -285,28 +324,38 @@ private static X509ChainStatusFlags MapVerifyErrorToChainStatus(Interop.libcrypt
             }
         }
 
-        internal static X509Certificate2Collection FindCandidates(
+        internal static List<X509Certificate2> FindCandidates(
             X509Certificate2 leaf,
-            X509Certificate2Collection extraStore)
+            X509Certificate2Collection extraStore,
+            List<X509Certificate2> downloaded,
+            ref TimeSpan remainingDownloadTime)
         {
-            X509Certificate2Collection candidates = new X509Certificate2Collection();
+            List<X509Certificate2> candidates = new List<X509Certificate2>();
             Queue<X509Certificate2> toProcess = new Queue<X509Certificate2>();
             toProcess.Enqueue(leaf);
 
-            using (var rootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
-            using (var intermediateStore = new X509Store(StoreName.CertificateAuthority, StoreLocation.LocalMachine))
+            using (var systemRootStore = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
```

Comment:
I think this has reached the point where it makes more sense to create a List<> of X509Store's and then use a loop to execute the same logic on all of them.